Classes/Service/NodeRedirectService.php

```diff
@@ -303,14 +306,23 @@ protected function createUriPathsAcrossDimensionsForNode(string $nodeIdentifier,
      */
     protected function hasNodeUriChanged(NodeInterface $node, Workspace $targetWorkspace): bool
     {
-        $newUriPath = $this->buildUriPathForNode($node);
-        $newUriPath = $this->removeContextInformationFromRelativeNodeUri($newUriPath);
-
         $nodeInTargetWorkspace = $this->getNodeInWorkspace($node, $targetWorkspace);
         if (!$nodeInTargetWorkspace) {
             return false;
         }
-        $oldUriPath = $this->buildUriPathForNode($nodeInTargetWorkspace);
+        try {
+            $newUriPath = $this->buildUriPathForNode($node);
+        } catch (\Exception $exception) {
+            $this->logger->info(sprintf('Failed to build new URI for updated node "%s": %s', $node->getContextPath(), $exception->getMessage()));
+            return false;
+        }
```

Comment:
^ I moved that block below the `$nodeInTargetWorkspace` check. Otherwise it was always executed even if no node was updated at all (performance optimization)